### PR TITLE
Sema: Fix crash with observers vs invalid 'override' attribute

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes <rdar://problem/57040259>.

### CODE REVIEW

File: validation-test/compiler_crashers_2_fixed/rdar57040259.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// RUN: not %target-swift-frontend -typecheck %s
+
+class A {}
+
+class B: A {
+  func foo(_: () -> ()) {
+    override var prop: Any? {
+      didSet {}
```
Comment (hughbe) [inline validation-test/compiler_crashers_2_fixed/rdar57040259.swift]: Could also add test for willSet

**Comment by DougGregor** [comment]
Hah: https://github.com/apple/swift/pull/28446

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by theblixguy** [comment]
Maybe we can close #28446?

**Comment by DougGregor** [comment]
Yeah, this is why I didn't go with the solution Slava chose :)
```
16:55:13 /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/multifile/synthesized-accessors/materialize-for-set-1/Inputs/library.swift:11:21: note: through reference here
16:55:13   open override var value: Int32 { didSet { } }
16:55:13                     ^
16:55:13 /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/multifile/synthesized-accessors/materialize-for-set-1/Inputs/library.swift:11:21: note: through reference here
16:55:13   open override var value: Int32 { didSet { } }
16:55:13                     ^
```

**Comment by slavapestov** [comment]
@DougGregor thank you :)

**Pull request closed without merging**